### PR TITLE
Update java stub template to check path components before qualifying

### LINKED ISSUE #11186: java_stub_template improperly qualifies paths
In [java_stub_template.txt](https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt;drc=02f8da948bf3955304a4cef9399bd3907430bbc4;l=298), when building the classpath for the manifest file, every path entry is first qualified with the working directory, even when the path is already an absolute path.

The code should check for relative paths before qualifying them.

### PR DESCRIPTION
**Pull request by katre**
them with the pwd.

Fixes #11186.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt [hunk 1]
```diff
@@ -288,28 +288,41 @@ ARGS=(
...
     CLASSPATH_SEPARATOR=":"
-    URI_PREFIX="file:$(pwd)/"  # e.g. "file:/usr/local/foo.jar"
   fi
 
-  URI_PREFIX=${URI_PREFIX//\//\\/}
-  MANIFEST_CLASSPATH="${CLASSPATH_SEPARATOR}${CLASSPATH}"
+  OLDIFS="$IFS"
+  IFS="${CLASSPATH_SEPARATOR}" # Use a custom separator for the loop.
+  for path in ${CLASSPATH}; do
+    # Convert spaces to %20
```
Comment (lberki) [inline src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt:302]: very attentive :)

File: src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt [hunk 2]
```diff
@@ -288,28 +288,41 @@ ARGS=(
...
 
-  MANIFEST_CLASSPATH=$(sed "s/ /%20/g" <<< "${MANIFEST_CLASSPATH}")
-  MANIFEST_CLASSPATH=$(sed "s/$CLASSPATH_SEPARATOR/ $URI_PREFIX/g" <<< "${MANIFEST_CLASSPATH}")
+    if is_windows; then
+      path="file:/${path}" # e.g. "file:/C:/temp/foo.jar"
```
Comment (lberki) [inline src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt:305]: isn't this supposed to be `file://` (note the double slash?)

File: src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt [hunk 3]
```diff
@@ -288,28 +288,41 @@ ARGS=(
...
+    else
+      # If not absolute, qualify the path
+      case "${path}" in
+        /*) ;; # Already an absolute path
+        *) path="$(pwd)/${path}";; # Now qualified
+      esac
+      path="file:${path}" # e.g. "file:/usr/local/foo.jar"
+    fi
+
+    path=$(sed "s/ /%20/g" <<< "${path}")
```
Comment (lberki) [inline src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt:315]: (I keep forgetting that this syntax exists)

File: src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt (hunk 2 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt:305]: No, the syntax is, apparently, file:(path), where path can be relative to the current directory or absolute. This matches the previous implementation.

File: src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt (hunk 3 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt:315]: Me too.

**Pull request closed without merging**